### PR TITLE
Don't crash when viewing in progress recording.

### LINKED ISSUE #501: app exits/crashes to home screen when trying to play recording still in progress
**Describe the bug**
App crashes when attempting to play a recording in progress

**To Reproduce**
1. Start recording something currently playing or schedule a future recording (must use web or other app for this step)
2. Once the recording has started, open DVR library in roku app
3. navigate to the show, season, and episode
4. press Ok on remote to play
5. get returned to roku home screen

**Expected behavior**
Recording should play (or if this is not support for some reason, a clear error should be returned)

**Logs**
Nothing is logged on server side, at least at default log level.

**Screenshots**

**Additional context**
jellyfin server is docker 10.7.7
jellyfin roku app is version 1.4 build 9
RokuTV 7104X TCL Model 49S04
Software Version 10.5.0 build 4201-30

### PR DESCRIPTION
**Pull request by jimdogx**
Current version crashes when trying to view a recording in progress.

**Changes**
Defensive coding

**Issues**
Fixes #501

### CODE REVIEW

File: source/VideoPlayer.brs [hunk 1]
```diff
@@ -30,7 +30,7 @@ sub AddVideoContent(video, mediaSourceId, audio_stream_idx = 1, subtitle_idx = -
 
     ' Special handling for "Programs" or "Vidoes" launched from "On Now" or elsewhere on the home screen...
     ' basically anything that is a Live Channel.
-    if meta.json.ChannelId <> invalid
+    if isValid(meta.json) and isValid(meta.json.ChannelId)
```
Comment (1hitsong) [inline source/VideoPlayer.brs]:
```suggestion
    if isValid(meta?.json?.ChannelId)
```

File: source/VideoPlayer.brs [hunk 2]
```diff
@@ -175,7 +175,17 @@ sub AddVideoContent(video, mediaSourceId, audio_stream_idx = 1, subtitle_idx = -
     if mediaSourceId = invalid
         mediaSourceId = video.id
     end if
-    if meta.live then mediaSourceId = "" ' Don't send mediaSourceId for Live media
+
+    ' Don't send mediaSourceId for Live Media
+    ' Note: Recordings in progress will have meta.live = invalid, but we still don't want to send mediaSourceId
+    if meta.live = invalid
```
Comment (1hitsong) [inline source/VideoPlayer.brs]:
```suggestion
    if not isValid(meta.live)
```

File: source/VideoPlayer.brs [hunk 3]
```diff
@@ -197,7 +207,7 @@ sub AddVideoContent(video, mediaSourceId, audio_stream_idx = 1, subtitle_idx = -
 
     video.container = getContainerType(meta)
 
-    if m.playbackInfo.MediaSources[0] = invalid
+    if m.playbackInfo.MediaSources[0] = invalid and isValid(meta.json)
```
Comment (1hitsong) [inline source/VideoPlayer.brs]:
```suggestion
    if not isValid(m.playbackInfo.MediaSources[0]) and isValid(meta.json)
```

File: source/VideoPlayer.brs [hunk 4]
```diff
@@ -395,7 +405,7 @@ end function
 
 function getContainerType(meta as object) as string
     ' Determine the file type of the video file source
-    if meta.json.mediaSources = invalid then return ""
+    if meta.json = invalid or meta.json.mediaSources = invalid then return ""
```
Comment (1hitsong) [inline source/VideoPlayer.brs]:
```suggestion
    if not isValid(meta?.json?.mediaSources) then return ""
```

**Review by 1hitsong — CHANGES_REQUESTED** [review]

File: source/VideoPlayer.brs (hunk 1 above)
Comment (jimdogx) [reply source/VideoPlayer.brs]: I keep forgetting that Roku introduced the ? operator recently.  Thanks 👍

**Review by 1hitsong — APPROVED** [review]

**Review by neilsb — APPROVED** [review]

**Comment by numericOverflow** [comment]
@jimdogx - Is there a way for me to test this updated version of the Roku app now that you've implemented a fix?

**Comment by jimdogx** [comment]
@numericOverflow if you feel comfortable with side loading apps on your Roku (it does not remove or alter your main installation), you can read up on it more here:

https://github.com/jellyfin/jellyfin-roku/blob/unstable/DEVGUIDE.md

Feel free to ask questions in the [Matrix](https://matrix.to/#/+jellyfin:matrix.org) 😄

**Pull request merged**